contracts/stake/src/state.rs

```diff
@@ -314,11 +337,13 @@ impl StakeState {
             );
         }
 
-        if stake.reward == 0 {
+        // Shift eligibility only if warnings are saturated
+        if effective_faults > 0 {
             // stake.amount can be None if the provisioner unstake in the same
```

Comment:
The comment refers to `// stake.amount can be None if the provisioner unstake in the same`

Reply:
Oh ok
Well I tried to reuse the previous unwrapped var, but due to borrow checker I need to take it again from the option.
Then, instead of unwrapping, I `let some` and left the comment.

I'll try to find a better way

Reply:
Borrow checker satisfied, let me know